### DISCUSSION TITLE
Iceoryx lib and application depends on it in yocto

### DISCUSSION THREAD

**Discussion by PatilShubham-21** [eclipse-iceoryx/iceoryx#2005] [Q&A]
Hello Team,

One of my R&D I'm trying to integrate iceoryx libraries into yocto build system so I can build my application from yocto build system which has dependency on iceoryx.

I'm able to integrate iceoryx into my yocto build and it's generating all lib for hoofs, posh and platform etc. But when I tries to build my application or iceoryx sample application using yocto it's throwing below CMake error. In shot not able to find platform lib path.

CMake Error at /home/shubpat/data/yocto-tda4/build/arago-tmp-default-glibc/work/aarch64-oe-linux/icedelivery/1.0.0-r0/recipe-sysroot/usr/lib/cmake/iceoryx_platform/iceoryx_platformTargets.cmake:73 (message):
|   The imported target "iceoryx_platform::iceoryx_platform" references the
|   file
| 
|      "/usr/lib/libiceoryx_platform.so.2.90.0"
| 
|   but this file does not exist.  Possible reasons include:
| 
|   * The file was deleted, renamed, or moved to another location.
| 
|   * An install or uninstall procedure did not complete successfully.
| 
|   * The installation package was faulty and contained
| 
|      "/home/shubpat/data/yocto-tda4/build/arago-tmp-default-glibc/work/aarch64-oe-linux/icedelivery/1.0.0-r0/recipe-sysroot/usr/lib/cmake/iceoryx_platform/iceoryx_platformTargets.cmake"
| 
|   but not all the files it references.
| 
| Call Stack (most recent call first):
|   /home/shubpat/data/yocto-tda4/build/arago-tmp-default-glibc/work/aarch64-oe-linux/icedelivery/1.0.0-r0/recipe-sysroot/usr/lib/cmake/iceoryx_platform/iceoryx_platformConfig.cmake:55 (include)
|   CMakeLists.txt:24 (find_package)
| 
| 
| -- Configuring incomplete, errors occurred!

Can you tell me how can I resolve this lib path issue

**Comment by PatilShubham-21** [comment]
iceoryx-emaple yocto reciepe

SUMMARY = "A recipe for building iceoryx example"
LICENSE = "Apache"
LIC_FILES_CHKSUM = "file://${WORKDIR}/git/LICENSE;md5=46d6aa0ba1fa2ed247cd8d42f20b72f4"

BRANCH = "master"
SRCREV = "414784e169ba58666a9b4c0be087ea3e043ae839"
SRC_URI = "git://github.com/eclipse-iceoryx/iceoryx.git;protocol=https;branch=${BRANCH}"

DEPENDS = "iceoryx cpptoml"

inherit cmake

S = "${WORKDIR}/git/iceoryx_examples/icedelivery"

EXTRA_OECMAKE = " \
		-DCMAKE_PREFIX_PATH=${libdir} \
		-DCMAKE_MODULE_PATH=${libdir} \
		-DCMAKE_INSTALL_PREFIX=${exec_prefix} \
    "

===================================================================

iceoryx lib building yocto recipes

SUMMARY = "A recipe for creating iceoryx library and memory management daemon."
LICENSE = "Apache"
LIC_FILES_CHKSUM = "file://${WORKDIR}/git/LICENSE;md5=46d6aa0ba1fa2ed247cd8d42f20b72f4"

BRANCH = "master"
SRCREV = "414784e169ba58666a9b4c0be087ea3e043ae839"
SRC_URI = "git://github.com/eclipse-iceoryx/iceoryx.git;protocol=https;branch=${BRANCH}"

DEPENDS = "acl cpptoml"
RDEPENDS:${PN} = ""

RPROVIDES:${PN} = "iox-roudi"

inherit cmake

S = "${WORKDIR}/git/iceoryx_meta"

EXTRA_OECMAKE = " \
		-DCMAKE_PREFIX_PATH=${libdir} \
		-DBUILD_SHARED_LIBS=ON \
		-DDOWNLOAD_TOML_LIB=OFF \
		-DCMAKE_INSTALL_BINDIR=${bindir} \
		-DCMAKE_INSTALL_SBINDIR=${sbindir} \
		-DCMAKE_INSTALL_LIBDIR=${libdir} \
		-DCMAKE_INSTALL_INCLUDEDIR=${includedir} \
		-DCMAKE_INSTALL_DATAROOTDIR=${datadir} \
    "

SYSROOT_DIRS += "${bindir}"

# Include additionally generated default configs
FILES:${PN} += "/usr/etc/*"

**Comment by elfenpiff** [comment]
@PatilShubham-21 

Is the file `/usr/lib/libiceoryx_platform.so.2.90.0` actually present? You can check this with `ls /usr/lib/libiceoryx_platform.so.2.90.0`.

If the file is not present you forgot to install iceoryx, I think the simplest way is to call
```
cd build
sudo make install
```

if I am not mistaken.

**Comment by PatilShubham-21** [comment]
Hi @elfenpiff,

Thanks for support...!

Yes I have installed iceoryx lib. you can see in below pic. The Important thing is I'm trying all these thing in yocto build system not in normal cmake/make build.

![image](https://github.com/eclipse-iceoryx/iceoryx/assets/56061141/7e6c838f-85dc-4d06-96fc-8f40d7707d27)


ERROR is in do_configure(i.e cmake command)

![image](https://github.com/eclipse-iceoryx/iceoryx/assets/56061141/5915f7c8-e719-45d0-b65e-d92b3c466a48)

**Reply by elBoberido** [reply]
@PatilShubham-21 it might be an issue with the install prefix. It looks like cmake is looking and `/usr/...` but it is installed in `/home/.../work/...`. You need to tell cmake where to look for the files. This can be done by specifying the `DCMAKE_PREFIX_PATH`, e.g. `-DCMAKE_PREFIX_PATH=/home/.../recipe-sysroot`
